### PR TITLE
feat: Try all nodes and allow backup nodes

### PR DESCRIPTION
**Pull request by ckniffen**
## High Level Overview of Change

s1 and s2 stopped responding. This will allow the explorer to try multiple entry points at once and choose the winner.

### Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)

### CODE REVIEW

**Comment by ckniffen** [comment]
Still need to add some tests.

File: src/containers/shared/SocketContext.tsx [hunk 1]
```diff
@@ -23,7 +23,19 @@ function getSocket(rippledUrl?: string): XrplClient {
...
+    process.env.VITE_RIPPLED_SECONDARY.split(',').forEach((network) => {
+      wsUrls.push.apply(wsUrls, [
+        `${prefix}://${network}:${process.env.VITE_RIPPLED_WS_PORT}`,
+        `${prefix}://${network}:443`,
+      ])
+    })
+  }
+
+  const socket = new XrplClient(wsUrls, {
+    tryAllNodes: true,
```
Comment (mvadari) [inline src/containers/shared/SocketContext.tsx:37]: Wasn't this causing problems? Were those fixed?

**Review by mvadari — APPROVED** [review]

File: src/containers/shared/SocketContext.tsx (hunk 1 above)
Comment (ckniffen) [reply src/containers/shared/SocketContext.tsx:35]: They were fixed in 2.0.3

File: src/containers/shared/SocketContext.tsx [hunk 2]
```diff
@@ -12,25 +12,37 @@ function isInsecureWs(rippledHost: string | undefined): boolean {
 }
 
 function getSocket(rippledUrl?: string): XrplClient {
```
Comment (JST5000) [inline src/containers/shared/SocketContext.tsx:14]: Should `getSocket` accept an optional string array of urls instead of being converted to an array immediately?

File: src/containers/shared/test/SocketContext.test.ts [hunk 3]
```diff
@@ -19,18 +19,47 @@ describe('getSocket', () => {
 
   describe('server defined entrypoint', () => {
     beforeEach(() => {
+      delete process.env.VITE_RIPPLED_SECONDARY
```
Comment (JST5000) [inline src/containers/shared/test/SocketContext.test.ts:22]: Why is this necessary? (Maybe worth a comment?)

Comment (JST5000) [reply src/containers/shared/test/SocketContext.test.ts:22]: Does this have side effects if run on production? (i.e. is the process.env value permanently removed?)

Comment (ckniffen) [reply src/containers/shared/test/SocketContext.test.ts:22]: I will remove this.  This was originally there because I was going to have two environment variables.  Since all entry points are tried at once it made no sense for them to be called secondary.

File: src/containers/shared/SocketContext.tsx (hunk 2 above)
Comment (ckniffen) [reply src/containers/shared/SocketContext.tsx:14]: I kept it this way to make the handling of the url as an environment variable or user specified work the same.

File: src/containers/shared/test/SocketContext.test.ts (hunk 3 above)
Comment (mvadari) [reply src/containers/shared/test/SocketContext.test.ts:22]: First off, this is a test, so it's not run in production. Second off, there's a `beforeEach` in the outside `describe` that saves and replaces `process.env`.

**Review by JST5000 — APPROVED** [review]
LGTM
